src/sql/sqlops.d.ts

```diff
@@ -108,6 +108,68 @@ declare module 'sqlops' {
 		}
 	}
 
+	/**
+	 * Namespace for interacting with Object Explorer
+	*/
+	export namespace objectexplorer {
+		/**
+		 * Get an Object Explorer node corresponding to the given connection and path. If no path
+		 * is given, it returns the top-level node for the given connection. If there is no node at
+		 * the given path, it returns undefined.
+		 * @param {string} connectionId The id of the connection that the node exists on
+		 * @param {string?} nodePath The path of the node to get
+		 * @returns {ObjectExplorerNode} The node corresponding to the given connection and path,
+		 * or undefined if no such node exists.
+		*/
+		export function getNode(connectionId: string, nodePath?: string): Thenable<ObjectExplorerNode>;
```

Comment:
From an API consumer's point of view, it's not clear how I'm supposed to get a hold of `nodePath`. After some trial and error, I managed to find the right format and use the API successfully, but I've run into a couple of oddities in the process (in this example, the path was `localhost,1401/Databases/WideWorldImporters/Tables/Application.Cities (System-Versioned)`):

1. The first segment of the path needs to be the server name, which is a bit redundant given I'm already providing the `connectionId` -- I'd expect the path to be _relative_ to the connection.
2. I need to know the folder structure of the node I want to access, and the node's path is coupled to the text used for display purposes -- for example, I'd expect the last segment of the path to be `Application.Cities` instead of `Application.Cities (System-Versioned)`. Ideally, I'd like to select an object by specifying its database, schema and object names. However, I understand that might be too high-level, so maybe it'd be nice if there were factory methods (maybe on `ObjectExplorerNodePath`) that allowed building paths to common objects, and then allowing navigation from there:

    ```typescript
    const path = ObjectExplorerNodePath.forTable('WideWorldImporters', 'Application', 'Cities');
    const citiesTable = await getNode(connectionId, path);
    const primaryKey = await citiesTable.getChild('Keys/PK_Application_Cities');
    ```
    or
    ```typescript
    const path = ObjectExplorerNodePath
        .forTable('WideWorldImporters', 'Application', 'Cities')
        .join('Keys/PK_Application_Cities');
    const primaryKey = await getNode(connectionId, path);
    ```

Reply:
Thanks for taking a look at this and for your feedback. I agree that building node paths as an API user isn't great. After this I'll start working on something like the `ObjectExplorerNodePath` factory you're suggesting